### ISSUE TITLE
Binary Releases are Dynamically Linked to glibc

### ISSUE THREAD

**Issue by berney** [josenk/terraform-provider-esxi#30]
The binary release of this plugin is dynamically linked to glibc.

When I try to run `terraform` on Alpine Linux I get this error:

```
terraform plan

Error: Error asking for user input: 2 error(s) occurred:

* module.esxi0.provider.esxi: fork/exec /root/.terraform.d/plugins/terraform-provider-esxi_v1.4.0: no such file or directory   
```

The file exists and is executable.

```
ls -l ~/.terraform.d/plugins/terraform-provider-esxi_v1.4.0 
-rwxr-xr-x    1 root     root      29819820 Feb 12 00:53 /root/.terraform.d/plugins/terraform-provider-esxi_v1.4.0
```

Looking at the file you can see it is dynamically linked.

```
file ~/.terraform.d/plugins/terraform-provider-esxi_v1.4.0
/root/.terraform.d/plugins/terraform-provider-esxi_v1.4.0: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, Go BuildID=VI2SYs4Q2VMNXpchizXh/IrW4Ik3CjPUbzSBm3BHO/yQ-rMDQpiIyKniLAqoBd/mid7UKwhB1WKuHLsfWlC, with debug_info, not stripped
```

It is dynamically linked to glibc

```
ldd ~/.terraform.d/plugins/terraform-provider-esxi_v1.4.0
        /lib64/ld-linux-x86-64.so.2 (0x7efbac846000)
        libpthread.so.0 => /lib64/ld-linux-x86-64.so.2 (0x7efbac846000)
        libc.so.6 => /lib64/ld-linux-x86-64.so.2 (0x7efbac846000)
```

**Comment by josenk** [comment]
There is just way too many distros for me to support.   I have been including a binary that should work for most linux users, but it looks like it doesn't work for you.    I've included build instructions in the README.   Building your own binary will be the most compatible and optimal for your system.

**Comment by silasb** [comment]
We might be able to statically compile the binary.  I've used this successfully with another Go program that was linking to linuxbrew glibc that was newer than my target host.  Obviously there are some downsides like larger binaries.  We could offer it as an additional binary.

**Comment by berney** [comment]
The point I meant to make was to statically compile rather than dynamically link. Its a larger binary, but in this day and age I don't think it matters as much, the binary should then be pretty universal and work on just about every linux distro regardless of libc differences etc (e.g. Alpine using muscl over glibc).

**Comment by silasb** [comment]
Can you try building like this: `CGO_ENABLED=0 GOOS=linux go build` and seeing if it'll work on Alpine?

**Comment by josenk** [comment]
I'm doing regression tests...   I'll most likely be switching to static binaries in the next release.   In the mean time, you should be able to build your own static binary by using above options.

**Comment by josenk** [comment]
Should be fixed in v1.4.1
>https://github.com/josenk/terraform-provider-esxi/releases/tag/1.4.1

**Comment by berney** [comment]
This is working well, thanks for making the change and the swift action.

This is my Dockerfile

```Dockerfile
FROM hashicorp/terraform
ARG ESXIPROV_VER=1.4.1
ADD https://github.com/josenk/terraform-provider-esxi/releases/download/${ESXIPROV_VER}/terraform-provider-esxi_v${ESXIPROV_VER} /root/.terraform.d/plugins/
RUN chmod +x /root/.terraform.d/plugins/terraform-provider-esxi_v${ESXIPROV_VER}
COPY . /tph/terraform
WORKDIR /tph/terraform
```

The only problem is ovftool, I haven't yet managed to get it installed properly, I need to copy an already installed version. When I get it working I'll send a PR.
